packages/next/pages/_document.tsx

```diff
@@ -479,6 +479,17 @@ export class Head extends Component<
                 href={canonicalBase + getAmpPath(ampPath, dangerousAsPath)}
               />
             )}
+            {isFallback && (
+              <link
+                rel="preload"
+                href={
+                  assetPrefix + `/_next/data/${buildId}${dangerousAsPath}.json`
```

Comment:
Is `dangerousAsPath` correct here? This would result in the incorrect data bundle for rewritten dynamic pages AFAIK.

Reply:
We probably want `pathname` (href) w/ dynamic route parts interpolated.

Reply:
Let's be sure to add a test case for this, one way or another. The test should render a fallback page by means of a rewrite, then find this preload in the DOM and request it to verify it's real.

Reply:
It looks like we might not be able to use `link[rel=preload]` to kick-off the data fetching in the fallback since the data URL needs to be specific to the dynamic route e.g. `/blog/post-1` and we can only specify one since we're pre-rendering it